### PR TITLE
Add support for Balena by detecting the engine in local sync and usin…

### LINKED ISSUE #144: Balena needs to be identified to avoid breaking local sync
This: https://github.com/resin-io-modules/resin-sync/blob/master/lib/sync/local-resin-os-device.coffee#L79 checks for a `docker.pid`, which with balena is replaced by balena.pid.
So sync should either do:
```
$(if [ -f '/var/run/balena.pid' ]; then cat /var/run/balena.pid; else cat /var/run/docker.pid; fi)
```
Or use `docker.version()` as in docker-progress: https://github.com/resin-io/docker-progress/blob/master/index.coffee#L241 to identify balena and look for the correct file.

### PR DESCRIPTION
**Pull request by pcarranzav**
…g the correct pid file

Closes #144 
Change-Type: minor
Signed-off-by: Pablo Carranza Velez <pablo@resin.io>

### CODE REVIEW

File: lib/docker-utils.coffee [hunk 1]
```diff
@@ -335,4 +335,9 @@ class RdtDockerUtils
 					else
 						throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 
+	isBalena: =>
+		@docker.versionAsync().get('Engine')
+		.then (engine) ->
+			return engine == 'balena'
```
Comment (lekkas) [inline lib/docker-utils.coffee:341]: In https://github.com/resin-io/docker-progress/blob/master/index.coffee#L34 there is a check for `balaena` as well - do we have any released resinOS versions with `balaena` as the engine version?

**Review by lekkas — APPROVED** [review]
LGTM, my only question is on balena/balaena

File: lib/docker-utils.coffee (hunk 1 above)
Comment (pcarranzav) [reply lib/docker-utils.coffee:341]: Not that I know of - balena was introduced into resinOS only a few days ago https://github.com/resin-os/meta-resin/pull/902 so even if a previous version identified itself as "balaena", only the one that identifies itself as "balena" would have a balena.pid, anything else would still use docker.pid

**Review by pimterry — APPROVED** [review]

**Comment by pcarranzav** [comment]
Thanks @lekkas and @pimterry  - Tested on different resinOS versions, one with balena and one with docker. Works fine afaict. Merging

**Pull request merged**